lib/tasks/api_token.rake

```diff
@@ -0,0 +1,18 @@
+namespace :api_token do
+  namespace :lead_provider do
+    desc "Generate a new API token for the Lead Providers API"
+    task :generate_token, %i[lead_provider_name_or_id] => :environment do |_t, args|
+      logger = Rails.env.test? ? Rails.logger : Logger.new($stdout)
+
+      lead_provider_name_or_id = args.lead_provider_name_or_id
```

Comment:
althought pretty small maybe we can move this into a service, test the service and the rake task can accept a lead provider or loop through all lead providers otherwise?